src/osd/PGLog.cc

```diff
@@ -1016,3 +1016,74 @@ void PGLog::rebuild_missing_set_with_deletes(
 
   set_missing_may_contain_deletes();
 }
+
+#ifdef WITH_SEASTAR
+seastar::future<> PGLog::rebuild_missing_set_with_deletes(
+  crimson::os::FuturizedStore *store,
+  crimson::os::CollectionRef ch,
+  const pg_info_t &info)
+{
+  return seastar::do_with(map<hobject_t, pg_missing_item>(),
+    [this, info, store, ch] (auto& extra_missing) {
+    // save entries not generated from the current log (e.g. added due
+    // to repair, EIO handling, or divergent_priors).
+    for (auto& [oid, item] : missing.take_items()) {
+      if (!log.logged_object(oid)) {
+	extra_missing[oid] = std::move(item);
```

Comment:
please reconsider the comment of https://github.com/ceph/ceph/pull/33943#discussion_r393550377